### PR TITLE
feat(wifi): read wifi state on non-englisch systems

### PR DESCRIPTION
**Pull request by Khaos66**
### Prerequisites

- [x] I have read and understood the `CONTRIBUTING` guide
- [x] The commit message follows the [conventional commits][cc] guidelines
- [ ] Tests for the changes have been added (for bug fixes/features)
- [ ] Docs have been added / updated (for bug fixes/features)

### Description

This fixes issue #1405 and makes the `wifi` segment work on non-englisch systems.
It's kind of a mess though. I used this article a lot:
https://medium.com/@justen.walker/breaking-all-the-rules-using-go-to-call-windows-api-2cbfd8c79724

Feel free to suggest changes to clean this up a little ;)

### CODE REVIEW

File: src/environment.go [hunk 1]
```diff
@@ -105,6 +120,7 @@ type environmentInfo interface {
 	cache() cache
 	close()
 	logs() string
+	getWifiNetworks() (*wifiInfo, error)
```
Comment (JanDeDobbeleer) [inline src/environment.go]: there doesn't seem to be a unix/mac alternative implemented here

File: src/segment_wifi.go [hunk 2]
```diff
@@ -47,8 +37,19 @@ func (w *wifi) enabled() bool {
 		return false
 	}
 
-	// Extract data from netsh cmdResult
-	parseNetshCmdResult(cmdResult, w)
+	if wifiInfo == nil {
+		return false
+	}
+
+	w.Connected = wifiInfo.Connected
```
Comment (JanDeDobbeleer) [inline src/segment_wifi.go]: If they're able to be mapped ,why aren't we aligning the names?

**Comment by JanDeDobbeleer** [comment]
@Khaos66 should I have a look at this one?

**Comment by Khaos66** [comment]
@JanDeDobbeleer  I'm taking a short break over the holidays ;) you can take over if you'd like to.

**Comment by JanDeDobbeleer** [comment]
@Khaos66 can you allow edits from maintainers for this PR?

**Comment by Khaos66** [comment]
Sure. But how? A link to the docs would be great 😅

**Comment by JanDeDobbeleer** [comment]
https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/working-with-forks/allowing-changes-to-a-pull-request-branch-created-from-a-fork

**Comment by Khaos66** [comment]
Looks like this checkbox is already set. Can you not commit to the branch?

**Comment by JanDeDobbeleer** [comment]
@Khaos66 I couldn't validate from the mobile app, in that case I can probably just push! Thanks 🙏🏻

**Comment by JanDeDobbeleer** [comment]
@Khaos66 I need someone to validate my changes as I don't have the ability to setup a wifi connection on my VM. 
@mikesigs I removed the `State` property as I don't see the case of displaying this when you're not connected.

**Comment by Khaos66** [comment]
@JanDeDobbeleer sure thing. Will do so soon ;)

**Comment by Khaos66** [comment]
@JanDeDobbeleer Good job at cleaning this mess up! But I have bad news: It's not working =/
I added comments to the changes you made ;)

Actually, finding the correct types for all the fields of these win32 structs was hell =D so plz don't break it by removing unused fields. They are needed, so the returned win32 api structs match the go structs in memory. Else the values of the fields will not be correct.

**Comment by JanDeDobbeleer** [comment]
@Khaos66 gotcha. It's just that I need to mark them for the linter to ignore in that case. Will fix.

**Comment by JanDeDobbeleer** [comment]
@Khaos66 I reverted the renames

**Comment by Khaos66** [comment]
@JanDeDobbeleer Now it's working again <3

**Comment by JanDeDobbeleer** [comment]
In that case I'm merging 👌🏻

**Comment by mikesigs** [comment]
Nice work!

**Pull request merged**